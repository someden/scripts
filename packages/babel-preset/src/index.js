const {
	declare
} = require('@babel/helper-plugin-utils');
const getEnvOptions = require('./envs');
const preset = require('./preset');

module.exports = declare(
	(api, options) => preset(api, getEnvOptions(options.env), options)
);
