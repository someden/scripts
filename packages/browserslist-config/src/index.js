module.exports = [].concat(
	require('./browsers'),
	require('./node')
);
