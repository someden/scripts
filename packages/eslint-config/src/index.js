/**
 * Combine all configs without ReactJS config
 */

module.exports = {
	extends: [
		'eslint:recommended',
		'plugin:jsdoc/recommended'
	].concat([
		'./rules/common',
		'./rules/es6',
		'./rules/jsdoc'
	].map(require.resolve)),
	env: {
		es6: true
	},
	parser: 'babel-eslint',
	parserOptions: {
		ecmaVersion: 6,
		sourceType: 'module',
		ecmaFeatures: {
			impliedStrict: true,
			experimentalObjectRestSpread: true,
			objectLiteralDuplicateProperties: false
		}
	}
};
