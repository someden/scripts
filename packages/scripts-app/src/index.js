import path from 'path';
import update from 'immutability-helper';
import {
	FILL_ME,
	getScriptAndArgs,
	getScriptArg,
	pushArgs
} from '@trigen/scripts/helpers/args';

export const Features = [
	'eslint',
	'tslint'
];

export const DefaultFeatures = [
	'tslint'
];

const storybookConfigs = path.join(__dirname, 'configs', 'storybook');
const scripts = {
	'lint:styles':     {
		cmd:  'stylelint',
		args: FILL_ME
	},
	'lint:js':    {
		cmd:  'eslint',
		args: FILL_ME
	},
	'lint:ts':    {
		cmd:  'tslint',
		args: FILL_ME
	},
	'lint:scripts':    FILL_ME,
	'lint':            ['lint:styles', 'lint:scripts'],
	'typecheck':       {
		cmd:  'tsc',
		args: ['--noEmit', '--pretty', '--skipLibCheck']
	},
	'jest':            {
		vars: { NODE_ENV: 'test' },
		cmd:  'jest',
		args: ['-c', 'jest.config.json']
	},
	'test':            ['typecheck', 'lint', 'jest', 'build'],
	'build:docs':      [{
		cmd:          'typedoc',
		args:         FILL_ME,
		ignoreResult: true
	}, {
		cmd:       'touch',
		args:      ['docs/.nojekyll'],
		immutable: true
	}],
	'start:storybook': {
		vars: {},
		cmd:  'start-storybook',
		args: FILL_ME
	},
	'build:storybook': {
		vars: { NODE_ENV: 'development' },
		cmd:  'build-storybook',
		args: FILL_ME
	},
	'start':           {
		vars: { NODE_ENV: 'development' },
		cmd:  'node',
		args: [path.join(__dirname, 'start.js')]
	},
	'build':           {
		vars: { NODE_ENV: 'production' },
		cmd:  'node',
		args: [path.join(__dirname, 'build.js')]
	},
	'serve':           {
		vars: { NODE_ENV: 'production' },
		cmd:  'node',
		args: [path.join(__dirname, 'serve.js')]
	}
};
const scriptsNames = Object.keys(scripts);

export default function getScripts(inputArgs, features) {

	const {
		script,
		args
	} = getScriptAndArgs(inputArgs, scriptsNames);
	const storybookConfigsArgs = getScriptArg(args, '-c', storybookConfigs);
	const withCustomSotrybookConfigs = !storybookConfigsArgs.length;
	const scriptsWithArgs = update(scripts, {
		'lint:styles':     {
			args: {
				$set: getScriptArg(args, 0, 'src/**/*.css')
			}
		},
		'lint:js':         {
			args: {
				$set: [
					'--cache',
					...getScriptArg(args, 0, 'src/**/*.{js,jsx}')
				]
			}
		},
		'lint:ts':         {
			args: {
				$set: [
					'-p', '.', '-t', 'stylish',
					...getScriptArg(args, 0, 'src/**/*.{ts,tsx}')
				]
			}
		},
		'lint:scripts':    {
			$set: [
				features.eslint && 'lint:js',
				features.tslint && 'lint:ts'
			].filter(Boolean)
		},
		'build:docs':      {
			0: {
				args: {
					$set: [
						'./src',
						...getScriptArg(args, '--out', './docs'),
						'--excludeExternals', '--mode', 'modules'
					]
				}
			}
		},
		'start:storybook': {
			vars: {
				CUSTOM_CONFIGS: {
					$set: JSON.stringify(withCustomSotrybookConfigs)
				}
			},
			args: {
				$set: [
					'--ci', '-p', '3001',
					...storybookConfigsArgs
				]
			}
		},
		'build:storybook': {
			vars: {
				CUSTOM_CONFIGS: {
					$set: JSON.stringify(withCustomSotrybookConfigs)
				}
			},
			args: {
				$set: [
					...storybookConfigsArgs,
					'-o', 'storybook-build'
				]
			}
		}
	});

	pushArgs(scriptsWithArgs, args);

	return {
		scripts: scriptsWithArgs,
		exec:    script
	};
}
