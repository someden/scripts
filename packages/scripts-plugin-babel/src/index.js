import update from 'immutability-helper';
import {
	FILL_ME,
	getScriptArg
} from '@trigen/scripts/helpers';

const scripts = {
	start: {
		vars: {
			NODE_ENV: 'development'
		},
		cmd: 'babel-node',
		args: FILL_ME
	},
	build: {
		vars: {
			NODE_ENV: 'production'
		},
		cmd: 'babel',
		args: FILL_ME
	}
};

export default function getScripts(args, allScripts) {
	return update(allScripts, {
		start: {
			$set: update(scripts.start, {
				args: {
					$push: [
						...getScriptArg(args, 0, 'src/index.js'),
						...args
					]
				}
			})
		},
		build: {
			$set: update(scripts.build, {
				args: {
					$push: [
						...getScriptArg(args, 0, './src'),
						...getScriptArg(args, '-d', './lib'),
						...getScriptArg(args, '-s', 'inline'),
						...args
					]
				}
			})
		}
	});
}
