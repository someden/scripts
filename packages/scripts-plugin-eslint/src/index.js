import update from 'immutability-helper';
import {
	FILL_ME,
	getScriptArg,
	addScripts
} from '@trigen/scripts/helpers';

const scripts = {
	'lint:js':      {
		cmd:  'eslint',
		args: FILL_ME
	},
	'lint:scripts': ['lint:js'],
	'lint':         ['lint:scripts'],
	'test':         ['lint']
};

export default function getScripts(args, allScripts, {
	lint = 'src/**/*.{js,jsx}'
} = {}) {
	return update(allScripts, {
		'lint:js':      {
			$set: update(scripts['lint:js'], {
				args: {
					$push: [
						'--cache',
						...getScriptArg(args, 0, lint),
						...args
					]
				}
			})
		},
		'lint:scripts': {
			$apply: _ => addScripts(_, scripts['lint:scripts'])
		},
		'lint':         {
			$apply: _ => addScripts(_, scripts.lint)
		},
		'test':         {
			$apply: _ => addScripts(_, scripts.test)
		}
	});
}
