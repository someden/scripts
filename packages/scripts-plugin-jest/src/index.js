import path from 'path';
import update from 'immutability-helper';
import {
	FILL_ME,
	getScriptArg,
	addScripts
} from '@trigen/scripts/helpers';

const scripts = {
	'jest': {
		vars: { NODE_ENV: 'test' },
		cmd:  'jest',
		args: FILL_ME
	},
	'artifacts': {
		cmd:  'node',
		args: [path.join(__dirname, 'artifacts.js')]
	},
	'test':  ['jest'],
	'eject': {
		cmd:  'cp',
		args: [
			path.join(__dirname, 'artifacts.js'),
			'scripts/artifacts.js'
		]
	}
};

export default function getScripts(args, allScripts) {
	return update(allScripts, {
		'jest': {
			$set: update(scripts.jest, {
				args: {
					$push: [
						...getScriptArg(args, '-c', 'jest.config.json'),
						...args
					]
				}
			})
		},
		'artifacts': {
			$set: update(scripts.artifacts, {
				args: {
					$push: args
				}
			})
		},
		'test': {
			$apply: _ => addScripts(_, scripts.test)
		},
		'eject': {
			$apply: _ => addScripts(_, scripts.eject)
		}
	});
}
