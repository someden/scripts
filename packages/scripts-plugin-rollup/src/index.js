import update from 'immutability-helper';

const scripts = {
	'build': {
		vars: { NODE_ENV: 'production' },
		cmd:  'rollup',
		args: ['-c']
	}
};

export default function getScripts(args, allScripts) {
	return update(allScripts, {
		'build': {
			$set: update(scripts.build, {
				args: {
					$push: args
				}
			})
		}
	});
}
