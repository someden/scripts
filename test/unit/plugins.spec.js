import fs from 'fs';
import path from 'path';

const packages = path.join(process.cwd(), 'packages');
const plugins = fs.readdirSync(packages).filter(
	_ => _ !== 'scripts' && _[0] != '.'
);

describe('@trigen/scripts-*', () => {

	plugins.forEach((_) => {

		it(`${_} should return correct scripts object`, () => {

			expect(
				// eslint-disable-next-line import/no-dynamic-require
				require(path.join(packages, _, 'src', 'index.js')).default([], {})
			).toMatchSnapshot();
		});
	});
});
