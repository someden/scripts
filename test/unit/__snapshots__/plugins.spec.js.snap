// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@trigen/scripts-* scripts-plugin-babel should return correct scripts object 1`] = `
"{
	\\"start\\": {
		\\"vars\\": {
			\\"NODE_ENV\\": \\"development\\"
		},
		\\"cmd\\": \\"babel-node\\",
		\\"args\\": [
			\\"src/index.js\\"
		]
	},
	\\"build\\": {
		\\"vars\\": {
			\\"NODE_ENV\\": \\"production\\"
		},
		\\"cmd\\": \\"babel\\",
		\\"args\\": [
			\\"./src\\",
			\\"-d\\",
			\\"./lib\\",
			\\"-s\\",
			\\"inline\\"
		]
	}
}"
`;

exports[`@trigen/scripts-* scripts-plugin-eslint should return correct scripts object 1`] = `
"{
	\\"lint:js\\": {
		\\"cmd\\": \\"eslint\\",
		\\"args\\": [
			\\"--cache\\",
			\\"src/**/*.{js,jsx}\\"
		]
	},
	\\"lint:scripts\\": [
		\\"lint:js\\"
	],
	\\"lint\\": [
		\\"lint:scripts\\"
	],
	\\"test\\": [
		\\"lint\\"
	]
}"
`;

exports[`@trigen/scripts-* scripts-plugin-jest should return correct scripts object 1`] = `
"{
	\\"jest\\": {
		\\"vars\\": {
			\\"NODE_ENV\\": \\"test\\"
		},
		\\"cmd\\": \\"jest\\",
		\\"args\\": [
			\\"-c\\",
			\\"jest.config.json\\"
		]
	},
	\\"test\\": [
		\\"jest\\"
	]
}"
`;

exports[`@trigen/scripts-* scripts-plugin-rollup should return correct scripts object 1`] = `
"{
	\\"build\\": {
		\\"vars\\": {
			\\"NODE_ENV\\": \\"production\\"
		},
		\\"cmd\\": \\"rollup\\",
		\\"args\\": [
			\\"-c\\"
		]
	}
}"
`;

exports[`@trigen/scripts-* scripts-plugin-size-limit should return correct scripts object 1`] = `
"{
	\\"test\\": [
		\\"checkSize\\"
	],
	\\"checkSize\\": {
		\\"cmd\\": \\"size-limit\\",
		\\"args\\": []
	}
}"
`;

exports[`@trigen/scripts-* scripts-plugin-storybook should return correct scripts object 1`] = `
"{
	\\"start:storybook\\": {
		\\"vars\\": {
			\\"STORYBOOK_AUTO_CONFIGURE\\": \\"true\\"
		},
		\\"cmd\\": \\"start-storybook\\",
		\\"args\\": [
			\\"--ci\\",
			\\"-p\\",
			\\"3001\\",
			\\"-c\\",
			\\"/packages/scripts-plugin-storybook/src/storybook\\"
		]
	},
	\\"build:storybook\\": {
		\\"vars\\": {
			\\"NODE_ENV\\": \\"development\\",
			\\"STORYBOOK_AUTO_CONFIGURE\\": \\"true\\"
		},
		\\"cmd\\": \\"build-storybook\\",
		\\"args\\": [
			\\"-c\\",
			\\"/packages/scripts-plugin-storybook/src/storybook\\",
			\\"-o\\",
			\\"storybook-build\\"
		]
	}
}"
`;

exports[`@trigen/scripts-* scripts-plugin-typescript should return correct scripts object 1`] = `
"{
	\\"lint:ts\\": {
		\\"cmd\\": \\"tslint\\",
		\\"args\\": [
			\\"-p\\",
			\\".\\",
			\\"-t\\",
			\\"stylish\\",
			\\"src/**/*.{ts,tsx}\\"
		]
	},
	\\"lint:scripts\\": [
		\\"lint:ts\\"
	],
	\\"lint\\": [
		\\"lint:scripts\\"
	],
	\\"typecheck\\": {
		\\"cmd\\": \\"tsc\\",
		\\"args\\": [
			\\"--noEmit\\",
			\\"--pretty\\",
			\\"--skipLibCheck\\"
		]
	},
	\\"test\\": [
		\\"typecheck\\",
		\\"lint\\"
	],
	\\"start\\": {
		\\"vars\\": {
			\\"NODE_ENV\\": \\"development\\"
		},
		\\"cmd\\": \\"ts-node-dev\\",
		\\"args\\": [
			\\"-P\\",
			\\"./tsconfig.dev.json\\",
			\\"-r\\",
			\\"tsconfig-paths/register\\",
			\\"--respawn\\",
			\\"--transpileOnly\\",
			\\"--ignore-watch\\",
			\\"node_modules\\",
			\\"src/index.ts\\"
		]
	},
	\\"build:docs\\": [
		{
			\\"cmd\\": \\"typedoc\\",
			\\"args\\": [
				\\"./src\\",
				\\"--out\\",
				\\"./docs\\",
				\\"--ignoreCompilerErrors\\",
				\\"--excludeExternals\\",
				\\"--mode\\",
				\\"modules\\"
			],
			\\"ignoreResult\\": true
		},
		{
			\\"cmd\\": \\"touch\\",
			\\"args\\": [
				\\"docs/.nojekyll\\"
			]
		}
	],
	\\"build\\": {
		\\"vars\\": {
			\\"NODE_ENV\\": \\"production\\"
		},
		\\"cmd\\": \\"tsc\\",
		\\"args\\": [
			\\"--rootDir\\",
			\\"./src\\",
			\\"--outDir\\",
			\\"./lib\\"
		]
	}
}"
`;

exports[`@trigen/scripts-* scripts-preset-lib should return correct scripts object 1`] = `
"{
	\\"test\\": [
		\\"build\\"
	],
	\\"cleanPublish\\": [
		\\"test\\",
		{
			\\"cmd\\": \\"clean-publish\\",
			\\"args\\": []
		}
	]
}"
`;

exports[`@trigen/scripts-* scripts-preset-node-app should return correct scripts object 1`] = `
"{
	\\"start\\": {
		\\"vars\\": {
			\\"NODE_ENV\\": \\"development\\"
		},
		\\"cmd\\": \\"babel-node\\",
		\\"args\\": [
			\\"src/index.js\\"
		]
	},
	\\"build\\": {
		\\"vars\\": {
			\\"NODE_ENV\\": \\"production\\"
		},
		\\"cmd\\": \\"rollup\\",
		\\"args\\": [
			\\"-c\\"
		]
	},
	\\"jest\\": {
		\\"vars\\": {
			\\"NODE_ENV\\": \\"test\\"
		},
		\\"cmd\\": \\"jest\\",
		\\"args\\": [
			\\"-c\\",
			\\"jest.config.json\\"
		]
	},
	\\"test\\": [
		\\"jest\\",
		\\"build\\"
	]
}"
`;

exports[`@trigen/scripts-* scripts-preset-react-app should return correct scripts object 1`] = `
"{
	\\"start\\": {
		\\"vars\\": {
			\\"NODE_ENV\\": \\"development\\"
		},
		\\"cmd\\": \\"node\\",
		\\"args\\": [
			\\"/packages/scripts-preset-react-app/src/start.js\\"
		]
	},
	\\"build\\": {
		\\"vars\\": {
			\\"NODE_ENV\\": \\"production\\"
		},
		\\"cmd\\": \\"node\\",
		\\"args\\": [
			\\"/packages/scripts-preset-react-app/src/build.js\\"
		]
	},
	\\"lint:ts\\": {
		\\"cmd\\": \\"tslint\\",
		\\"args\\": [
			\\"-p\\",
			\\".\\",
			\\"-t\\",
			\\"stylish\\",
			\\"src/**/*.{ts,tsx}\\"
		]
	},
	\\"lint:scripts\\": [
		\\"lint:ts\\"
	],
	\\"lint\\": [
		\\"lint:styles\\",
		\\"lint:scripts\\"
	],
	\\"typecheck\\": {
		\\"cmd\\": \\"tsc\\",
		\\"args\\": [
			\\"--noEmit\\",
			\\"--pretty\\",
			\\"--skipLibCheck\\"
		]
	},
	\\"test\\": [
		\\"typecheck\\",
		\\"lint\\",
		\\"jest\\",
		\\"build\\"
	],
	\\"build:docs\\": [
		{
			\\"cmd\\": \\"typedoc\\",
			\\"args\\": [
				\\"./src\\",
				\\"--out\\",
				\\"./docs\\",
				\\"--ignoreCompilerErrors\\",
				\\"--excludeExternals\\",
				\\"--mode\\",
				\\"modules\\"
			],
			\\"ignoreResult\\": true
		},
		{
			\\"cmd\\": \\"touch\\",
			\\"args\\": [
				\\"docs/.nojekyll\\"
			]
		}
	],
	\\"jest\\": {
		\\"vars\\": {
			\\"NODE_ENV\\": \\"test\\"
		},
		\\"cmd\\": \\"jest\\",
		\\"args\\": [
			\\"-c\\",
			\\"jest.config.json\\"
		]
	},
	\\"start:storybook\\": {
		\\"vars\\": {
			\\"STORYBOOK_AUTO_CONFIGURE\\": \\"false\\",
			\\"REACT_APP_STORYBOOK_AUTO_CONFIGURE\\": \\"true\\"
		},
		\\"cmd\\": \\"start-storybook\\",
		\\"args\\": [
			\\"--ci\\",
			\\"-p\\",
			\\"3001\\"
		]
	},
	\\"build:storybook\\": {
		\\"vars\\": {
			\\"NODE_ENV\\": \\"development\\",
			\\"STORYBOOK_AUTO_CONFIGURE\\": \\"false\\",
			\\"REACT_APP_STORYBOOK_AUTO_CONFIGURE\\": \\"true\\"
		},
		\\"cmd\\": \\"build-storybook\\",
		\\"args\\": [
			\\"-o\\",
			\\"storybook-build\\"
		]
	},
	\\"lint:styles\\": {
		\\"cmd\\": \\"stylelint\\",
		\\"args\\": [
			\\"src/**/*.css\\"
		]
	},
	\\"serve\\": {
		\\"vars\\": {
			\\"NODE_ENV\\": \\"production\\"
		},
		\\"cmd\\": \\"node\\",
		\\"args\\": [
			\\"/packages/scripts-preset-react-app/src/serve.js\\"
		]
	}
}"
`;
